### ISSUE TITLE
CRD's in helm 3 do not support templating

### ISSUE THREAD

**Issue by rio** [traefik/traefik-helm-chart#441]
The changes in 9.20.0 basically break all helm 3 installs as the apiserver tries to parse the golang templates in the CRD as json since they aren't templated out.

```
Error: failed to install CRD crds/ingressroute.yaml: error parsing : json: line 0: invalid character '{' looking for beginning of object key string
```

**Comment by mohankumarbm** [comment]
$ helm install traefik traefik/traefik
Error: failed to install CRD crds/ingressroute.yaml: error parsing : json: offset 2: invalid character '{' looking for beginning of object key string

**Comment by rio** [comment]
After some testing with kind and different versions of kubernetes (v1.21, v1.16, v1.15) and different versions of helm (v3.6, v3.2) I've come to the conclusion that helm will not check if a version of an api is available before it tries to install that CRD. This means that coming kubernetes v1.22 when the `apiextensions.k8s.io/v1beta1` api is removed version <9.20.0 of this helm chart will also no longer install.

That's why I propose to revert the change made in 9.20.0 and remove the templating in the CRD's only leaving `apiextensions.k8s.io/v1beta1` to remain compatible with k8s <v1.22 and release that chart as 9.20.1.

Next replace all `apiextensions.k8s.io/v1beta1` CRD's with `apiextensions.k8s.io/v1` and release that chart as 10.0.0 marking it incompatible with k8s <v1.15.

For now I've had to pin all our manifests to traefik 9.19.2 because all our automated test deployment pipelines are failing :(

**Comment by rio** [comment]
As a side note, the helm website mentions the use of a separate chart for CRD management. This might be a way to continue to support helm charts after 10.0.0 on k8s <v1.16 by letting people install the CRD chart before installing the newer chart with the `--skip-crds` flag.

```bash
helm install deprecated-crds traefik/deprecated-crds
helm install traefik-without-crd traefik/traefik --version 10.0.0
```

**Comment by RaJiska** [comment]
I can confirm I'm having the same issue here:

```
$ helm install traefik traefik/traefik
Error: failed to install CRD crds/ingressroute.yaml: error parsing : json: offset 2: invalid character '{' looking for beginning of object key string
```

**Comment by SantoDE** [comment]
Let me take a look

**Comment by jakubhajek** [comment]
Hello All, 

We have confirmed that Helm v3 not allowing templating in the CRDs is the cause of this issue, and will be releasing an update today to resolve it.

In the meantime, we recommend pinning your chart versions to 9.19.2 until you have a chance to review our update.
**We apologize** for the issues, this caused, and are looking into ways to prevent broken CRDs from affecting installs moving forward.

Thank you

**Comment by Juneezee** [comment]
Will the upcoming update include bumping traefik to [v2.4.9](https://github.com/traefik/traefik/releases/tag/v2.4.9)?

**Comment by jakubhajek** [comment]
@Juneezee yes, it is already included in PR #443.
